test/helpers/Range.js

```diff
@@ -0,0 +1,19 @@
+// from https://stackoverflow.com/questions/42213522/mocking-document-createrange-for-jest
+
+const createContextualFragment = (html) => {
+  const div = document.createElement('div');
+  div.innerHTML = html;
+  return div.children[0]; // so hokey it's not even funny
+};
+
+function Range() {};
+
+Range.prototype.setEnd = () => {};
+Range.prototype.setStart = () => {};
+Range.prototype.getBoundingClientRect = () => {};
+Range.prototype.getClientRects = () => {};
+Range.prototype.createContextualFragment = createContextualFragment;
+
+// HACK: Polyfil that allows codemirror to render in a JSDOM env.
+global.window.document.createRange = () => new Range();
+global.Range = Range;
```

Comment:
I know this is a copy-n-paste job, but we could use a class for Range and remove the note about code mirror in comments